packages/x-charts/src/context/HighlightedProvider/HighlightedContext.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { SeriesId } from '../../models/seriesType/common';
+
+/**
+ * The data of the highlighted item.
+ * To highlight an item, you need to provide the series id and the item id.
+ * If targeting the whole series, you can omit the item id.
+ * To clear the highlight, set the value to an empty object.
+ *
+ * @example
+ * // Highlight the item with the series id 'london' and the item id 0.
+ * { seriesId: 'london', itemId: 0 }
+ *
+ * // Highlight the whole series with the series id 'london'.
+ * { seriesId: 'london' }
+ *
+ * // Clear the highlight.
+ * {}
+ */
+export type HighlightItemData = {
+  /**
+   * The series id of the highlighted item.
+   */
+  seriesId?: SeriesId;
+  /**
+   * The item id of the highlighted item. Usually, it is the index of the data.
+   */
+  itemId?: number | string;
```

Comment:
Do you mean replacing `HighlightItemData` with `PieItemIdentifier|BarItemIdentifier|...`? I didn't want to tie them together because the idea is to eventually add `value` to `HighlightItemData` to allow for highlighting everything with the same effective value.

Or do you mean allowing multiple highlights?

```
[
  {
     seriesId: 'a',
     itemId: 'item1'
  },
  {
     seriesId: 'a',
     itemId: 'item2'
  }
]
```

which would be an interesting option 🤔

Reply:
I was thinking about something like in a treview of a disk space

```js
{ 
  seriesId: "diskSpace",
  path: ["home", "dev", "mui", "mui-x"]
}
```

to highlight the mui-x folder.

My point is that for now you can only itdentify items from it's data index which does nto work if you have nested structure

Reply:
unless we force that every step has an "id" property, which could be a bit annoying 😅

Reply:
We could use something like how `get` in lodash works https://youmightnotneed.com/lodash#get and allow for both strings and array.

Reply:
Sorry, I badly express my opinion 🙈. I was not asking to support review out of the box


My concern was not about supporting any specific series. 

Before each series had its own typing for identifying an element, and this type was used

https://github.com/mui/mui-x/blob/91cbec22c295450b547d8ed1f71cf845ced63fcf/packages/x-charts/src/hooks/useInteractionItemProps.ts#L41-L45

With this configuration, you can create a new series with whatever identifier you want. To do so you only need to add its type to `SeriesItemIdentifier`.

With this modification, we lose this freedom since every item is identified by its series' id and dataIndex.

I was wondering if this could become an issue in the future. From the different solution you proposed, it seems doable :+1:


> I've moved it to path?: string | string[], but right now there is not much use for string[]

Yes, from what I've seen it's only numbers that you are transforming to string with `.toString()`. Maybe better to use `dataIndex?: number. When deeper selector will be needed we could still extend `HighlightItemData` with

```
HighlightItemData = { seriesId, dataIndex } |  { seriesId, path }
```

Reply:
Tbh I would lean more towards a single solution if it works best. `path` solves the issue by having a single accessing pattern, though is a bit less descriptive when the path is a single value. Though `/` is still called a path in directories 😆 

We could allow `number | string | (string|number)[]` which would simplify it for users using only `dataIndex` referenced values, but it also introduces a bit more complexity/parsing if they are using it to index their own dataset.

Another option would be to have `{ seriesId, dataindex, path }` but again, the user would also need to know which is used when.

Reply:
> the user would also need to know which is used when.

We could add a parameter to distinguish them.

```
{ seriesId, dataIndex: number, hasNestedStructure?: false }
  | { seriesId, path: number[], hasNestedStructure: true }
```

which would allow to do 

```js
if(item.hasNestedStructure){
  // TS know path exist and dataindex don't
}
```


----

> Though `/` is still called a path in directories

Yes, it's naming a property according to its edge case instead of the main one.

IMO, a user seeing this JSDocs will say

1. What's the difference between dataindex and index of the data
2. why do they call it path if its an index

```ts
/**
   * The path to access the data.
   * - On Cartesian charts, the path is the dataIndex of the series.
   * - On Pie charts, the path is the index of the data.
   */
  path?: string | string[];
```

If you want to keep it as a single typing. WHich has the benefit of being simpler to document in API pages, we could name according to the main usecase

```ts
/**
   * The index of the element in the series data.
   */
  dataIndex?: number;
```

And when needed extend it

```ts
/**
   * The index of the element in the series data.
   * Can be an array if the series has a nested structure.
   */
  dataIndex?: number | number[];
```

Reply:
My point is that `path/dataPath` is a single mental model. You want to get the item at `0`? `path: 0`, you want to get the item at `[0].x`, `path: '[0].x'`

I'm not convinced this is the best way, but we can go with `dataIndex: number` while we dont have other use-cases, so we can take a better informed decision in the future 👍